packages/xo-web/src/xo-app/xostor/index.js

```diff
@@ -0,0 +1,559 @@
+const formatDiskName = name => '/dev/' + name
+const isDiskHaveChildren = disk => Array.isArray(disk.children) && disk.children.length > 0
+const isDiskGoodType = disk => disk.type === 'disk' || /^raid/.test(disk.type)
+const isDiskNotMounted = disk => disk.mountpoint === ''
+const isDiskNotRo = disk => disk.ro === '0'
+const isGoodNumberOfHosts = hosts => size(hosts) >= N_HOSTS_MIN && size(hosts) <= N_HOSTS_MAX
+const isXcpngPool = host => host?.productBrand === 'XCP-ng'
+const rendererDiskSelect = disk => (
+  <span>
+    <Icon icon='disk' /> {formatDiskName(disk.name)} {formatSize(Number(disk.size))}
+  </span>
+)
+const xostorDiskPredicate = disk =>
+  isDiskGoodType(disk) && isDiskNotRo(disk) && isDiskNotMounted(disk) && !isDiskHaveChildren(disk)
+
+// ===================================================================
+
+const StorageCard = decorate([
+  injectState,
+  ({ effects, state }) => (
+    <Card>
+      <CardHeader>{_('storage')}</CardHeader>
+      <CardBlock>
+        <Row>
+          <Col size={6}>
+            {_('name')}
+            <DebounceInput className='form-control' name='srName' onChange={effects.linkState} value={state.srName} />
+          </Col>
+          <Col size={6}>
+            {_('description')}
+            <DebounceInput
+              className='form-control'
+              name='srDescription'
+              onChange={effects.linkState}
+              value={state.srDescription}
+            />
+          </Col>
+        </Row>
+      </CardBlock>
+    </Card>
+  ),
+])
+
+const SettingsCard = decorate([
+  provideState({
+    computed: {
+      showWarningReplication: state => state.replication?.value === 1,
+    },
+  }),
+  injectState,
+  ({ effects, state }) => (
+    <Card>
+      <CardHeader>{_('settings')}</CardHeader>
+      <CardBlock>
+        <div>
+          <label>
+            <strong>{_('replication')}</strong>
+          </label>
+          <Select options={REPLICATION_OPTIONS} onChange={effects.onReplicationChange} value={state.replication} />
+          {state.showWarningReplication && (
+            <p className='text-warning'>
+              <Icon icon='alarm' /> {_('xostorReplicationWarning')}
+            </p>
+          )}
+        </div>
+        <div className='form-group'>
+          <label>
+            <strong>{_('provisioning')}</strong>
+          </label>
+          <Select onChange={effects.onProvisioningChange} options={PROVISIONING_OPTIONS} value={state.provisioning} />
+        </div>
+      </CardBlock>
+    </Card>
+  ),
+])
+
+const PoolCard = decorate([
+  connectStore({
+    srs: createGetObjectsOfType('SR').groupBy('$pool'),
+  }),
+  provideState({
+    initialState: () => ({ onlyShowXostorPools: true }),
+    effects: {
+      toggleState,
+    },
+    computed: {
+      poolPredicate: (state, props) => pool => {
+        if (!state.onlyShowXostorPools) {
+          return true
+        }
```

Comment:
Hm, maybe I am missing something, but predicates are directly passed to a `.filter()` function.
https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-web/src/common/select-objects.js#L354

Returning the value true instead of a function that returns true won't work.
```js
[...].filter(true)
// Uncaught TypeError: true is not a function

Reply:
I tried, but it doesn't work.
The `predicate` prop needs to be a function.

Reply:
Ok, I thought you could pass `true`, but I guess you can pass `undefined` then?